lib/extra/list_extra.mli

```diff
@@ -20,3 +20,8 @@ val max_on : ('a -> int) -> 'a list -> int
     accumulator. *)
 val map_with_fold :
   f:('a -> 'acc -> 'b * 'acc) -> init:'acc -> 'a list -> 'b list
+
+(** [map_with_fold_acc ~f ~init l] is just like [map_with_fold ~f ~init l] but
+    it returns the final value of the accumulator as well. *)
+val map_with_fold_acc :
```

Comment:
Let's rename to `map_and_fold`

Naming is hard and I know it looks similar to `map_with_fold`. But I still feel it's an improvement.

I'll rename when I come with better names 😅